### PR TITLE
Converts to using cf-core and BEM

### PR DESCRIPTION
**Pull request by himedlooff**
Testing the conversion to cf-core waters here, so far so good. :sailboat:

### CODE REVIEW

File: demo/custom.html [hunk 1]
```diff
@@ -3,66 +3,72 @@
 
 <!-- TEST: Vertical alignment of button links next to buttons. -->
```
Comment (Scotchester) [inline demo/custom.html:4]: Is it worth considering making the text of these comments visible (or available somehow) on the demo page? Might make it easier to know what the tests are trying to show if you're just viewing the demo.

Comment (himedlooff) [inline demo/custom.html:4]: Maybe? I'd suggest opening an issue for this if you want to recommend we do it.

**Comment by himedlooff** [comment]
One issue with converting to BEM is that `.btn-group` doesn't really follow it. I'd have to make `.btn-group_item` classes.

**Comment by Scotchester** [comment]
We could also ditch the wrapper if we wanted to do something like

``` css
.btn.btn__grouped__first
.btn.btn__grouped
.btn.btn__grouped__last
```

**Comment by Scotchester** [comment]
This will be a major breaking change for anyone already using this package, so we'll have to spread the word and warn them about updating.

**Comment by himedlooff** [comment]
I like that. I'm not sure about the double modifier thing what about:

``` css
.btn.btn__grouped-first
.btn.btn__grouped-middle
.btn.btn__grouped-last
```

**Comment by himedlooff** [comment]
Yeah should we bump this to 1.0.0 then?

**Comment by Scotchester** [comment]
I'm fine with that. `-middle` seems unnecessary, though.

**Comment by himedlooff** [comment]
Ok I'll ditch it.

**Comment by Scotchester** [comment]
These have been pretty well-tested; I'm good with calling it 1.0. If we do that, though, let's update the meta docs to the latest language found in [cfpb/open-source-project-template](https://github.com/cfpb/open-source-project-template) before merging.

**Comment by himedlooff** [comment]
Updated all of the OS stuff.

**Comment by himedlooff** [comment]
Alrighty, this is now using the updated docs as well.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
+
 # cf-buttons
 
 [![Bower dependency status](https://www.versioneye.com/user/projects/5345b61fe97a46bd58000150/badge.png)](https://www.versioneye.com/user/projects/5345b61fe97a46bd58000150)
-[![npm dependency status](https://gemnasium.com/cfpb/cf-typography.svg)](https://gemnasium.com/cfpb/cf-typography)
+[![npm dependency status](https://gemnasium.com/cfpb/cf-typography.svg)](https://gemnasium.com/cfpb/cf-buttons)
```
Comment (Scotchester) [inline README.md]: There's another instance of "typography" on this line, but I'll replace both of these with the unified Gemnasium badge after merging this.

Comment (Scotchester) [inline README.md]: Or you could do it. Here's the code:

``` markdown
[![Dependency Status](https://gemnasium.com/cfpb/cf-buttons.svg)](https://gemnasium.com/cfpb/cf-buttons)
```

File: README.md [hunk 3]
```diff
@@ -9,23 +10,43 @@ This component can be used by itself, but it was made for Capital Framework,
 a new front end framework developed at the
 [Consumer Financial Protection Bureau](https://cfpb.github.io/).
 
+- [View the docs](https://cfpb.github.io/cf-buttons/docs/)
+- [See the raw demo](https://cfpb.github.io/cf-buttons/demo/)
+
+The current version number can be found in [bower.json](bower.json#L3)
+and follows [semver](http://semver.org/).
```
Comment (Scotchester) [inline README.md]: If we're going to say this, let's be explicit about it and say "Semantic Versioning 2.0".

File: README.md [hunk 4]
```diff
@@ -9,23 +10,43 @@ This component can be used by itself, but it was made for Capital Framework,
...
+Release notes are recorded in the [releases page](https://github.com/cfpb/cf-buttons/releases/)
```
Comment (Scotchester) [inline README.md]: This is a little nitpicky, but can we say either "in the Releases section" or "on the Releases page"? And add a period at the end of the sentence.

File: README.md [hunk 5]
```diff
@@ -9,23 +10,43 @@ This component can be used by itself, but it was made for Capital Framework,
...
+
 If you would like to take advantage of more components or if you're new to
 Capital Framework, we encourage you to [start here](https://cfpb.github.io/capital-framework/).
 
-- [View the docs](https://cfpb.github.io/cf-buttons/docs/)
-- [See the raw demo](https://cfpb.github.io/cf-buttons/demo/)
+![](screenshot.png)
```
Comment (Scotchester) [inline README.md:24]: Did you mean to add a screenshot in here?

File: README.md [hunk 6]
```diff
@@ -9,23 +10,43 @@ This component can be used by itself, but it was made for Capital Framework,
...
 
-(Docs and demo built with the excellent [Topdoc](https://github.com/topcoat/topdoc/).)
 
+## How to get this running or how to use it
```
Comment (Scotchester) [inline README.md]: I wouldn't use this line from the template verbatim :)

How about, "How to use this component"?

File: TERMS.md [hunk 7]
```diff
@@ -38,7 +38,7 @@ all uses of the work, to the fullest extent permitted by applicable law.
 When using or citing the work, you should not imply endorsement by the
 author or the affirmer.
 
-[policy]: https://github.com/cfpb/source-code-policy/
+[policy]: http://github.com/cfpb/source-code-policy/
```
Comment (Scotchester) [inline TERMS.md]: Oops... looks like this needs to be edited in the template repo. Please revert this line to the `https` URL.

File: README.md (hunk 5 above)
Comment (Scotchester) [inline README.md:24]: Oops, NM, I'm dumb.

File: src/cf-buttons.less [hunk 8]
```diff
@@ -56,9 +59,7 @@
 
 .btn {
 
-    &,
-    &:link,
-    &:visited {
+    & {
```
Comment (Scotchester) [inline src/cf-buttons.less]: You could just bring these properties out of the `& { ... }` block now.

File: src/cf-buttons.less [hunk 9]
```diff
@@ -357,123 +359,138 @@
...
         border-top-right-radius: 0;
         border-bottom-right-radius: 0;
     }
 
-    .btn + .btn {
-        margin-left: unit( -3px / @btn-font-size, em );
-    }
-
-    .btn + .btn:not(:last-child) {
+    & {
```
Comment (Scotchester) [inline src/cf-buttons.less:464]: My comment about pulling this style out to the level above could apply here... but I also see why doing it this way might make it clearer in this case. So I'm fine with it here.

**Comment by himedlooff** [comment]
Thanks! I got them all.

**Comment by Scotchester** [comment]
:boom: Awesome!

**Comment by Scotchester** [comment]
Tag and release!

**Pull request merged**